source/opt/simplification_pass.cpp

```diff
@@ -71,8 +71,19 @@ bool SimplificationPass::SimplifyFunction(Function* function) {
               }
             });
             if (inst->opcode() == SpvOpCopyObject) {
-              context()->ReplaceAllUsesWith(inst->result_id(),
-                                            inst->GetSingleWordInOperand(0));
+              const auto replacement = inst->GetSingleWordInOperand(0);
+              std::unordered_set<Instruction*> uses_to_update;
+              get_def_use_mgr()->ForEachUse(
+                  inst, [&uses_to_update, replacement](Instruction* user,
+                                                       uint32_t operand) {
+                    const auto opcode = user->opcode();
+                    if (!spvOpcodeIsDebug(opcode) &&
```

Comment:
Suggest factoring into `ReplaceAllCodeUses` (or some better name).  We may need to do more of these.

Reply:
I refactored this into a RAUW version that takes a predicate. I left the predicate specific to this pass for now. I'm not sure we want to create a whole bunch of variants right away. Let me know what you think.

Reply:
Yup, this is good.  We can always add more predicates or make them globally available.  Thanks.